File: Common/InMotionGIT.Common/DataType/LookUpValueExtendCollection.cs
using System;
using System.Collections.ObjectModel;
using System.Data;
using System.Runtime.Serialization;
using System.Xml.Serialization;
using Microsoft.VisualBasic.CompilerServices;

namespace InMotionGIT.Common.DataType
{

    [CollectionDataContract(Namespace = "urn:InMotionGIT.Common.DataType")]
    [Serializable()]
    [XmlType(Namespace = "urn:InMotionGIT.Common.DataType")]
    [XmlRoot(Namespace = "urn:InMotionGIT.Common.DataType")]
    public class LookUpValueExtendCollection : Collection<LookUpValueExtend>
    {

        public LookUpValueExtend AddLookUpValue(int code, string description)
        {
            var item = new LookUpValueExtend(code.ToString(), description);
            Add(item);
            return item;
        }

        public LookUpValueExtend AddLookUpValue(int code, string description, string shortDescription)
        {
            var item = new LookUpValueExtend(code.ToString(), description, shortDescription);
            Add(item);
            return item;
        }

        public void LoadFromDataTable(DataTable tableInformation)
        {
            LookUpValueExtend item;
            foreach (DataRow row in tableInformation.Rows)
            {
                item = new LookUpValueExtend() { Code = Conversions.ToString(row["Code"]), Description = Conversions.ToString(row["Description"]), ShortDescription = Conversions.ToString(row["ShortDescription"]) };
                Add(item);
            }
        }

        public LookUpValueExtend GetItemByCode(int code)
        {
            LookUpValueExtend result = null;
            foreach (LookUpValueExtend Item in this)
            {
                if (Conversions.ToDouble(Item.Code) == code)
                {
                    result = Item;
                    break;
                }
            }
            return result;
        }

    }

}

File: Common/InMotionGIT.Common - Copy/DataType/LookUpValueExtendCollection.vb
Imports System.Collections.ObjectModel
Imports System.Runtime.Serialization
Imports System.Xml.Serialization

Namespace DataType

    <CollectionDataContract(Namespace:="urn:InMotionGIT.Common.DataType")>
    <Serializable()>
    <XmlType(Namespace:="urn:InMotionGIT.Common.DataType")>
    <XmlRoot(Namespace:="urn:InMotionGIT.Common.DataType")>
    Public Class LookUpValueExtendCollection
        Inherits Collection(Of LookUpValueExtend)

        Public Function AddLookUpValue(ByVal code As Integer, ByVal description As String) As LookUpValueExtend
            Dim item As New LookUpValueExtend(code, description)
            Me.Add(item)
            Return item
        End Function

        Public Function AddLookUpValue(ByVal code As Integer, ByVal description As String, shortDescription As String) As LookUpValueExtend
            Dim item As New LookUpValueExtend(code, description, shortDescription)
            Me.Add(item)
            Return item
        End Function

        Public Sub LoadFromDataTable(tableInformation As DataTable)
            Dim item As LookUpValueExtend
            For Each row As Data.DataRow In tableInformation.Rows
                item = New LookUpValueExtend With {.Code = row("Code"), .Description = row("Description"), .ShortDescription = row("ShortDescription")}
                Me.Add(item)
            Next
        End Sub

        Public Function GetItemByCode(code As Integer) As LookUpValueExtend
            Dim result As LookUpValueExtend = Nothing
            For Each Item As LookUpValueExtend In Me
                If Item.Code = code Then
                    result = Item
                    Exit For
                End If
            Next
            Return result
        End Function

    End Class

End Namespace